packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -110,14 +115,78 @@ export const updateAdditionalDepositData = async ({
     })
   }
 
-  // finally, else if the transaction is not ETH ie. it's a ERC20 token deposit
-  return updateTokenDepositStatusData({
+  let { value2 } = depositTx
+
+  // value2 not found in local storage, we fetch it instead
+  if (!value2) {
+    const _l1ToL2Msg = l1ToL2Msg as ParentToChildMessageReader
+    try {
+      // get maxSubmissionCost, which is the amount of ETH sent in batched ERC-20 deposit + max gas cost
+      const maxSubmissionCost = Number(
+        utils.formatEther(_l1ToL2Msg.messageData.maxSubmissionFee.toString())
+      )
+
+      // we deduct gas cost from max submission fee, which leaves us with amount2 (extra ETH sent with ERC-20)
+      if (depositTx.status === 'success') {
+        // if success, we use the actual gas cost
+        const gasCost = await getGasCostOnChildChain({
+          l1ToL2Msg: _l1ToL2Msg
+        })
+
+        if (!gasCost) {
+          throw 'failed to calculate gas cost, value2 will be undefined'
+        }
+
+        value2 = String(Number(maxSubmissionCost) - Number(gasCost))
+      } else {
+        // when not success, we don't know the final gas cost yet so we use estimates
+        const estimatedGasCost = utils.formatEther(
+          _l1ToL2Msg.messageData.gasLimit.mul(
+            _l1ToL2Msg.messageData.maxFeePerGas
+          )
+        )
+
+        value2 = String(Number(maxSubmissionCost) - Number(estimatedGasCost))
+      }
+
+      if (Number(value2) < MAX_SUBMISSION_FEE_THRESHOLD) {
+        // ETH amount too little to distinguish between gas used, won't show
+        value2 = undefined
+      }
+    } catch (_) {
+      value2 = undefined
+    }
+  }
+
+  const tokenDeposit = await updateTokenDepositStatusData({
     depositTx,
     l1ToL2Msg: l1ToL2Msg as ParentToChildMessageReader,
     timestampCreated,
     l1Provider,
     l2Provider
   })
+
+  // finally, else if the transaction is not ETH ie. it's a ERC20 token deposit
+  return {
+    ...tokenDeposit,
+    value2
+  }
+}
+
+const getGasCostOnChildChain = async ({
+  l1ToL2Msg
+}: {
+  l1ToL2Msg: ParentToChildMessageReader
+}) => {
+  const childReceipt = await l1ToL2Msg.getRetryableCreationReceipt()
```

Comment:
Hmm this is not the `childChainTxReceipt` but the creation receipt. 
For more accuracy - we should derive the child chain tx receipt by `ParentToChildMessageReader.getAutoRedeemAttempt()`. 
Check sdk for more info.

Reply:
updated